### DISCUSSION TITLE
Accel Calibration

### DISCUSSION THREAD

**Discussion by usbalbin** [HefnySco/OBAL#45] [Q&A]
Hi!

@Tjohoansson and I are now trying to calibrate the accelerometer, however it seems no matter what `AHRS_ORIENTATION`, one orientation is always reversed. We have the obal bord oriented with silkscreen up, pwm connectors in the rear, MPU-9250 to the left. The MPU-9250 has some arrows on it: X is to the left, Y is to the rear.

Reading the [doc](https://ardupilot.org/plane/docs/parameters.html#ahrs-orientation-board-orientation) it sounds like we should use `Yaw180`. That does, after Accel calibration, give correct readings for pitch, however roll is inverted.

Are we doing something wrong?

![PXL_20231018_163747224~2.jpg](https://github.com/HefnySco/OBAL/assets/10927717/8610fc76-abbf-47e1-b773-24d023821364)

![PXL_20231018_163737699~2.jpg](https://github.com/HefnySco/OBAL/assets/10927717/059149aa-9e3e-4167-bed1-a1bb84d18347)

**Comment by usbalbin** [comment]
Also, perhaps unrelated, we get warnings something about "main loop slow, 182/400Hz". Is there anything to do about that? cpu usage seems to be low for all cores.

**Comment by HefnySco** [comment]
> Hi!
> 
> @Tjohoansson and I are now trying to calibrate the accelerometer, however it seems no matter what `AHRS_ORIENTATION`, one orientation is always reversed. We have the obal bord oriented with silkscreen up, pwm connectors in the rear, MPU-9250 to the left. The MPU-9250 has some arrows on it: X is to the left, Y is to the rear.
> 
> Reading the [doc](https://ardupilot.org/plane/docs/parameters.html#ahrs-orientation-board-orientation) it sounds like we should use `Yaw180`. That does, after Accel calibration, give correct readings for pitch, however roll is inverted.
> 
> Are we doing something wrong?
> 
> ![PXL_20231018_163747224~2.jpg](https://user-images.githubusercontent.com/10927717/276334800-8610fc76-abbf-47e1-b773-24d023821364.jpg)
> 
> ![PXL_20231018_163737699~2.jpg](https://user-images.githubusercontent.com/10927717/276335278-059149aa-9e3e-4167-bed1-a1bb84d18347.jpg)

try to make 

> Hi!
> 
> @Tjohoansson and I are now trying to calibrate the accelerometer, however it seems no matter what `AHRS_ORIENTATION`, one orientation is always reversed. We have the obal bord oriented with silkscreen up, pwm connectors in the rear, MPU-9250 to the left. The MPU-9250 has some arrows on it: X is to the left, Y is to the rear.
> 
> Reading the [doc](https://ardupilot.org/plane/docs/parameters.html#ahrs-orientation-board-orientation) it sounds like we should use `Yaw180`. That does, after Accel calibration, give correct readings for pitch, however roll is inverted.
> 
> Are we doing something wrong?
> 
> ![PXL_20231018_163747224~2.jpg](https://user-images.githubusercontent.com/10927717/276334800-8610fc76-abbf-47e1-b773-24d023821364.jpg)
> 
> ![PXL_20231018_163737699~2.jpg](https://user-images.githubusercontent.com/10927717/276335278-059149aa-9e3e-4167-bed1-a1bb84d18347.jpg)

I believe you can use QGC and select auto orientation and it will give you the propoer orientation of the board.
